prog.go

```diff
@@ -251,9 +309,44 @@ type loadError struct {
 }
 
 func (le *loadError) Error() string {
-	return fmt.Sprintf("%s: %s", le.cause, le.verifierLog)
+	if le.verifierLog == "" {
+		return fmt.Sprintf("failed to load program: %s", le.cause)
+	}
+	return fmt.Sprintf("failed to load program: %s: %s", le.cause, le.verifierLog)
 }
 
 func (le *loadError) Cause() error {
 	return le.cause
 }
+
+func convertCString(in []byte) string {
+	inLen := bytes.IndexByte(in, 0)
+	return string(in[:inLen])
+}
+
+func checkName(name string) error {
```

Comment:
Would something like `regexp.MustCompile(fmt.Sprintf("^[0-9A-Za-z_]{0,%d}$", bpfObjNameLen))` be simpler?

Reply:
I guess it does, that should have occurred to me. This way we get better error messages though.